### ISSUE TITLE
[Enhance] Monetary Input to be Accessible

### ISSUE THREAD

**Issue by ssabo1** [sparkdesignsystem/Backlog#284]
**User Story:**
As Spark we want to update the $ used in our Monetary Input with Dollar Icon so that the input is accessible to users. 

The $ should be an SVG with `aria-hidden=true`, similar to the percentage input. This will hide the icon from screen readers.

`sprk-b-label--monetary` can probably be deleted.

This change will need to be made in Vanilla, Angular, and React.

**Notes:**
https://sparkdesignsystem.com/patterns/components/inputs#monetary-input

**AC:**
- Monetary Input will use $ Icon (replacing the text $)
- Monetary Input will be accessible
